src/kpatch_user.c

```diff
@@ -589,6 +593,93 @@ cmd_update(int argc, char *argv[])
 	return cmd_patch_user(ARRAY_SIZE(patch_all), patch_all);
 }
 
+#ifdef STRESS_TEST
+static int
+cmd_unpatch(int fd, int argc, char *argv[])
+{
+	int rv, pid;
+	char pid_str[64];
+	char *patch_pid_argv_unpatch[] = {
+		"unpatch",
+		"-p",
+		pid_str
+	};
+
+	rv = sscanf(argv[1], "%d", &pid);
+	if (rv != 1) {
+		kperr("can't parse pid from %s", argv[1]);
+		return -1;
+	}
+	sprintf(pid_str, "%d", pid);
+
+	optind = 1;
+	rv = cmd_unpatch_user(ARRAY_SIZE(patch_pid_argv_unpatch),
+			    patch_pid_argv_unpatch);
+	return rv;
+}
+
+static int
+server_wait(int pid, int period)
+{
+	kpatch_process_t _proc, *proc = &_proc;
+	struct timespec req, rem;
+	int rv, i;
+	req.tv_sec = 0;
+	req.tv_nsec = 1000*1000;
+	for (i=0; i<period; i++) {
+		nanosleep(&req, &rem);
+		rv = kpatch_process_init(proc, pid, 0, -1);
+		close(proc->fdmaps);
+		if (rv < 0) {
+			fprintf(stderr, "Process %d terminated.\n", pid);
+			return rv;
+		}
+	}
+	return 0;
+}
+
+static int
+server_stress_test(int fd, int argc, char *argv[])
+{
+	int rv,pid;
+	int period;
+	srand(time(NULL));
+
+	rv = sscanf(argv[1], "%d", &pid);
+	if (rv != 1) {
+		kperr("can't parse pid from %s", argv[1]);
+		return -1;
+	}
+
+	while (1) {
+		period = rand() % test_info.option_period;
+		if (server_wait(pid, period) < 0)
+			return 0;
+		if ((rv = cmd_execve_startup(fd, argc, argv, 0)) < 0) {
```

Comment:
While it should work with fd=-1 I think we should call the patch_user as it never sets `-s` argument which indicates that the process was just started. Also, we can hide away all the `-r` and `-s` options since they are not needed anymore.